presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/MapColumnValidator.java

```diff
@@ -44,16 +46,28 @@ public List<SingleColumn> generateChecksumColumns(Column column)
 
         Expression checksum = functionCall("checksum", column.getExpression());
         Expression keysChecksum = generateArrayChecksum(functionCall("map_keys", column.getExpression()), new ArrayType(keyType));
-        Expression valuesChecksum = generateArrayChecksum(functionCall("map_values", column.getExpression()), new ArrayType(valueType));
         Expression mapCardinalityChecksum = functionCall("checksum", functionCall("cardinality", column.getExpression()));
         Expression mapCardinalitySum = new CoalesceExpression(
                 functionCall("sum", functionCall("cardinality", column.getExpression())),
                 new LongLiteral("0"));
 
+        // We need values checksum in one case only: when key is a floating point type and value is not.
+        // In such case, when both checksum and the key checksum are bad, we cannot tell if the values match or not.
+        // Meaning we cannot resolve column mismatch, because value type is not a floating type an can mismatch.
+        // In order to be able to resolve column mismatch is such a situation, we want to generate extra checksum for the values.
```

Comment:
In order to resolve column mismatch in such a situation, generate an extra checksum for the values.